src/mame/drivers/vii.cpp

```diff
@@ -109,6 +109,8 @@
         lexizeus:
             Some corrupt sound effects and a few corrupt ground tiles a few minutes in. (checksum is good, and a video recorded
              from one of these doesn't exhibit these problems, so either emulation issue or alt revision?)
+        pvmil:
+            Question order is always the same from power-on / reset, it isn't on the real unit, what determines this?
```

Comment:
Can it really be considered working if the questions are completely predictable?  That seems game-breaking.

Reply:
well there are several things in MAME where 'random' orders aren't random because we're always starting from an identical state.  

I'm not really sure what determines the order here tho, doesn't seem to be any of the other port bits (as returning random on them doesn't change anything) and the game has no kind of NVRAM / storage.  The SoC has some kind of seeded RNG (that other things depend on) although I haven't checked if that is used, and the implementation of that was fixed by Moogly for the Disney games.

I'll admit, I was in 2 minds myself.

Reply:
I did wonder if it was some kind of 'rng running on clock' vs 'rng incremented on read' type issue, afaik in MAME even if you use the internal rng it isn't clock based.

Reply:
I've checked, and in this case the RNG does indeed depend upon the SunPlus SoC seeded RNG, and that is incremented on read.

Maybe the RNG there should also be being incremented on a timer?

Reply:
so I've put in a timer that will increment the RNG as well the increment on reads.  this would appear to be the only way the questions here can be random.

it's also something MAME probably needs to look at for random number generation in general (which IMHO should be a device, with selectable algorithms, when needed anyway)

Reply:
I strongly disagree with this change. The documentation says nothing about the PRNG registers being clocked at all, simply that they are reset to a known value and that writes will set specific values. I'm at work so I can't check right now, but perhaps these writes aren't currently implemented, and that's the underlying problem.
![prng](https://user-images.githubusercontent.com/1891617/67673838-4f67b100-f97b-11e9-868e-534dce0745e3.PNG)

Reply:
those writes are implemented.

the documentation doesn't state what clocks the polynomial.

if you clock it on read then you always get the same sequence, so question order is fixed

Reply:
Moogly says he's OK with this as long as I've tested the Disney puzzle games, which I have (I did prior to even making this submission as I knew they were sensitive to RNG stuff)